### ISSUE TITLE
Removing crags from app - eg copyright infringement

### ISSUE THREAD

**Issue by scd** [theCrag/mobile#20]
There are two scenarios that I can think of where we really don't want crags staying active in an app. Both of which potentially have serious safety or legal implications:

 - [ ] We are informed of a copyright infringement and have to take down some data. Easily done on the website. If the content lingers on peoples apps then this could cause real legal headaches.
 - [ ] There is some misleading information in a crag where following the information could lead to climbers getting themselves into a dangerous situation. Morally, once we know about this kind of scenario, we want to correct it as quickly as possible.

Both of these scenarios have happened in the last 6 months.

This makes me think that the initial version of the app must have a way of managing this. Nothing is yet implemented in the API, so we need to nut something out that is workable.

Initially I thought we could do callbacks, but they will not be in the first version. So I am thinking that the API returns a 'cragValidDate'. I presume that the app checks for index changes and crag updates on a regular basis (eg when the person logs in). If the cragValidDate is later then that stored in the the app then this crag is immediately removed from the app and downloaded again.

@nicHoch does this fit in with your design?

**Comment by brendanheywood** [comment]
Being devil's advocate, we can't do anything about a crag pdf, and we can't do anything about a crag app which is offline. We need to cater for both of those cases regardless, and that can only ever be through the T&C andother policies. I don't really see this as something that needs to be solved technically, but we will of course always provide a 'best effort' to keep content current regardless, which is just the normall process for updating data.

I don't really like the idea of some updates being forced, maybe over mobile data plans, while others are opt in and done by the user. I think it's just better if the user always updates explicitly. We could however flag to users that some updates contain critical updates and are more important.

**Comment by scd** [comment]
I agree with best effort, however I think best effort means updating immediately. Clearly if they are offline or have already printed then we cannot do anything about it. But there will only be a handful of these guys for a particular crag. However there may be 100,000 app installs.

I am thinking of the lawyer with the app saying you have copyright material on the app, then we say it is taken off but he looks at his app and it is still there. If you can action copyright infringements immediately then it hugely mitigates the damage. The copyright owner will not be too happy about their infringed material remaining in thousands of installs.

I am also thinking of the situation where the topo lines were not matching for some cases in the PDFs. I could not sleep until I invalidated all the PDFs that were effected.

I strongly disagree that we should not do anything technical here. It will not happen very often.
